### DISCUSSION TITLE
How can run aiokafka multi thread

### DISCUSSION THREAD

**Discussion by alm0ra** [aio-libs/aiokafka#979] [Q&A] [answered]
in this discussion in fast stream lib https://github.com/airtai/faststream/discussions/1222
The maintainer said `Aiokafka doesn't allow you to set up multiple connections in different threads`

Can you describe my problem and why it must run in the main thread?

**Comment by ods** [comment, marked as answer]
Each event loop in `asyncio` can be used in a single thread only, see [Python docs](https://docs.python.org/3/library/asyncio-dev.html#concurrency-and-multithreading) for explanation. So, you have to create dedicated event loop for each thread to run coroutines in them. Other that that I'm not aware of any obstacles to run `aikoafka` in multithreaded environment. But I've actually never checked that, as it's hard to imagine where it would give any benefit.
